cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -916,53 +916,59 @@ private Future<MapAndStatus<ConfigMap, KafkaRebalanceStatus>> onRebalancing(Reco
-                    JsonObject taskStatusJson = cruiseControlResponse.getJson();
-                    CruiseControlUserTaskStatus taskStatus = CruiseControlUserTaskStatus.lookup(taskStatusJson.getString("Status"));
-                    switch (taskStatus) {
-                        case COMPLETED:
-                            LOGGER.infoCr(reconciliation, "Rebalance ({}) is now complete", sessionId);
-                            p.complete(buildRebalanceStatus(
-                                    kafkaRebalance, null, KafkaRebalanceState.Ready, taskStatusJson, conditions));
-                            break;
-                        case COMPLETED_WITH_ERROR:
-                            // TODO: There doesn't seem to be a way to retrieve the actual error message from the user tasks endpoint?
-                            //       We may need to propose an upstream PR for this.
-                            // TODO: Once we can get the error details we need to add an error field to the Rebalance Status to hold
-                            //       details of any issues while rebalancing.
-                            LOGGER.errorCr(reconciliation, "Rebalance ({}) optimization proposal has failed to complete", sessionId);
-                            p.complete(buildRebalanceStatus(sessionId, KafkaRebalanceState.NotReady, conditions));
-                            break;
-                        case IN_EXECUTION: // Rebalance is still in progress
-                            LOGGER.infoCr(reconciliation, "Rebalance ({}) optimization proposal in execution", sessionId);
-                            // We need to check that the status has been updated with the ongoing optimisation proposal
-                            // The proposal field can be empty if a rebalance(dryrun=false) was called and the optimisation
-                            // proposal was still being prepared (in progress). In that case the rebalance will start when
-                            // the proposal is complete but the optimisation proposal summary will be missing.
-                            if (kafkaRebalance.getStatus().getOptimizationResult() == null ||
-                                    kafkaRebalance.getStatus().getOptimizationResult().isEmpty()) {
-                                LOGGER.infoCr(reconciliation, "Rebalance ({}) optimization proposal is now ready and has been added to the status", sessionId);
-                                p.complete(buildRebalanceStatus(
-                                        kafkaRebalance, sessionId, KafkaRebalanceState.Rebalancing, taskStatusJson, conditions));
-                            } else {
+                    if (cruiseControlResponse.getJson().isEmpty()) {
+                        // Cruise Control restarted: reset the state because the tasks queue is not persisted
+                        // this may also happen when the tasks' retention time expires, or the cache becomes full
```

Comment:
I would not strictly say the CC was restarted. Let's be generic about receiving an empty task queue from CC, or anyway the task we are asking for doesn't exist anymore. Of course CC restarted could be a reason but maybe not the only one.

Reply:
CC restart will be the most common cause, but I think the second line clarifies that it's not the only possible cause for this.

Reply:
so "Cruise Control restarted" should be used as a use case, not as a strong statement

Reply:
So something  like:
```
This may happen if:
1. Cruise Control restarted so resetting the state because the tasks queue is not persisted
2. Task's retention time expired, or the cache has become full
```

Reply:
fine with me

Reply:
Great, Thanks Paolo